client/src/actions/post.js

```diff
@@ -0,0 +1,95 @@
+import axios from 'axios'
+import { Toastr } from '../utils/toastr/Toastr'
+import { POSTS_END_LOADING, POSTS_RECIEVED, POSTS_START_LOADING } from '../utils/constants/actionsName'
+
+export const createPost = (uploadForm, setUploadForm) => {
+  const configMultipart = {
+    headers: {
+      'Content-Type': 'multipart/form-data'
+    }
+  }
+  const configJson = {
+    headers: {
+      'Content-Type': 'application/json'
+    }
+  }
+
+  const saveImgToStorageRequests = uploadForm.imagesToUpload.map((img, i) => {
+      const formData = new FormData()
+      formData.append("file", img.file)
+      return axios.post('/api/v1/storage/fake_upload', formData, configMultipart)
+        .catch(()=> {
+          uploadForm.imagesToUpload[i].uploadError = true
+          setUploadForm({...uploadForm})
+          return Promise.reject('Error during image ' + i + ' uploading')
+        })
+  })
+
+  Promise.all(saveImgToStorageRequests)
+    .then(result => {
+
+      const body = {
+        message: uploadForm.textToUpload,
+        image: result.map(response => response.data)[0],
+        showEveryone: true
+      }
+        return axios.post('/api/v1/posts', body, configJson)
+
+      },
+      reason => {
+
+      Toastr.error("One or more images were not uploaded")
+        return Promise.reject(reason)
+
+      })
+    .then(()=> {
+
+      setUploadForm({
+        imagesToUpload: [],
+        textToUpload: ''
+      })
+      window.location.reload()
+      Toastr.success("Your post was created")
+    })
+}
+
+export const getPostsForHomePage = () => async dispatch => {
+
+  dispatch({
+    type: POSTS_START_LOADING
```

Comment:
For what reason you don't use START/STOP_LOADING constants that we already have?

Reply:
Existing constants are for user authentication, they are check on ProtectedRouter stage. If you stick everything to them, you will see nothing (except spinner) on a page, until news feed is fully downloaded. Though, to my mind, it's better to show basic page (user avatar, some panels, etc) while downloading heavy news feed may take a little longer, so it needs parallel START/LOAD and separate spinner. Also think about infinity scroll in future. You don't want to hide whole page behind spinner while you just getting new bunch of posts from server, do you?

Reply:
Agree! Next step will be Start/Stop action with own ID ))